lib/memoizeWith/index.js

```diff
@@ -0,0 +1,10 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function memoizeWith(keyF, f) {
+  var cache = {}
+  return function() {
+    var key = keyF.apply(this, arguments)
+    cache[key] = cache[key] || { value: f.apply(this, arguments) }
```

Comment:
why do you write object into `cache[key]`?

and you could do shorter, something like this:

```js
var key = ...
return cache[key] || cache[key] = f.apply(this, arguments)
```

Reply:
```Uncaught ReferenceError: Invalid left-hand side in assignment```

Reply:
The only possible way is to do like below:

![image](https://user-images.githubusercontent.com/2991847/37898847-9206eb82-30f2-11e8-9b81-c9f2336a3fb9.png)

However, it's not larger or smaller. The same size.

Reply:
Yeah, I'm not accurate. Also you need to check `cache.hasOwnProperty(key)` because of `undefined` values

Reply:
Do we support `undefined`-keys?

Reply:
Hmm, maybe we should not do it.
And I think you should not define object as value, because why do you need it? And it will be shorter without it

Reply:
@BabkinAleksandr sometimes we need undefined values for some reasons, it's not difficult to support it

Reply:
So, checking `key in cache` could help
<img width="410" alt="index_js_ _memoizewith_ ___documents_nanoutils" src="https://user-images.githubusercontent.com/25172855/37900433-e9d82624-30f6-11e8-8082-4de147578a5b.png">

P.S. pass tests

Reply:
@BabkinAleksandr in your case `undefined` value effect in recalculating a function again, `counter` should be incorrect

Reply:
It passes your tests, so It works properly
<img width="431" alt="aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils__zsh_" src="https://user-images.githubusercontent.com/25172855/37903127-7acb55c2-30ff-11e8-97e9-652a0ae9f6c8.png">

Reply:
@BabkinAleksandr  you're right, it's not an array, blame on me